internal/powerplatform/services/connection/api_connection.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package powerplatform
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/google/uuid"
+	api "github.com/microsoft/terraform-provider-power-platform/internal/powerplatform/api"
+	powerplatform_helpers "github.com/microsoft/terraform-provider-power-platform/internal/powerplatform/helpers"
+)
+
+func NewConnectionsClient(api *api.ApiClient) ConnectionsClient {
+	return ConnectionsClient{
+		Api: api,
+	}
+}
+
+type ConnectionsClient struct {
+	Api *api.ApiClient
+}
+
+func (client *ConnectionsClient) BuildHostUri(environmentId string) string {
+	envId := strings.ReplaceAll(environmentId, "-", "")
+	realm := string(envId[len(envId)-2:])
+	envId = envId[:len(envId)-2]
+
+	return fmt.Sprintf("%s.%s.environment.%s", envId, realm, client.Api.GetConfig().Urls.PowerPlatformUrl)
+
+}
+
+func (client *ConnectionsClient) GetConnectorDefinition(ctx context.Context, environmentId, connectorName string) (*ConnectorDefinition, error) {
```

Comment:
This isn't referenced anywhere, what is the intended use case?

Reply:
fair point, removing